aws/resource_aws_cognito_user_pool_domain.go

```diff
@@ -22,10 +22,15 @@ func resourceAwsCognitoUserPoolDomain() *schema.Resource {
 
 		Schema: map[string]*schema.Schema{
 			"domain": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
```

Comment:
Can you please ensure `ValidateFunc: validateCognitoUserPoolDomain,` remains on the `domain` attribute? Thanks!

Reply:
Hi @bflad , thanks for your comments :-)
The problem here is that we don't know what to do in the validator context because the rules are differents if we have a custom domain or not and the other property of the resource is not accessible to determine the regex to use... So i decided to remove the validator and let the api throwing the error instead of terraform... The solution should be to validate the domain name against the two rules but that makes no sens...

Reply:
@adelamarre hmm, I was going off https://docs.aws.amazon.com/cognito-user-identity-pools/latest/APIReference/API_CreateUserPoolDomain.html still showing `Pattern: ^[a-z0-9](?:[a-z0-9\-]{0,61}[a-z0-9])?$` for domain. Is that not accurate anymore?

Reply:
If the `validateCognitoUserPoolDomain` usage here is being removed, then the function and its testing should be removed as well since it doesn't look like its used anywhere else.

Reply:
@bflad , No it's not accurate... as you can see the regex do not permit dots (.) in the name however there is always dots in a custom domain name...
Yes if the ```validateCognitoUserPoolDomain``` is not used anymore we could remove it and the affiliated test...

Reply:
Please note, on merge I am removing the now unused `validateCognitoUserPoolDomain` function and added 63 character string length validation back to the `domain` attribute as I encountered this error from the SDK/API in my testing:

```
--- FAIL: TestAccAWSCognitoUserPoolDomain_custom (5.54s)
    testing.go:538: Step 0 error: Error applying: 1 error occurred:
                * aws_cognito_user_pool_domain.main: 1 error occurred:
                * aws_cognito_user_pool_domain.main: Error creating Cognito User Pool Domain: InvalidParameterException: 1 validation error detected: Value 'tf-acc-test-domain-305748293312753547.--OMITTED--' at 'domain' failed to satisfy constraint: Member must have length less than or equal to 63
```